Tests/scripts/infrastructure_tests/lock_cloud_machines_test.py

```diff
@@ -1,7 +1,41 @@
+import json
+
 import pytest
+import requests
+from requests import ConnectionError, Response
 
 from Tests.scripts.lock_cloud_machines import get_my_place_in_the_queue, try_to_lock_machine,\
-    get_machines_locks_details, wait_for_build_to_be_first_in_queue, get_and_lock_all_needed_machines
+    get_machines_locks_details, wait_for_build_to_be_first_in_queue, get_and_lock_all_needed_machines, check_job_status
+
+
+@pytest.mark.parametrize(
+    "responses, expected_times_called, expected_status",
+    [
+        ([{'status': 'running'}], 1, 'running'),
+        ([ConnectionError, {'status': 'running'}], 2, 'running'),
+        ([ConnectionError, ConnectionError, {'status': 'running'}], 3, 'running'),
+        ([ConnectionError, ConnectionError, ConnectionError, {'status': 'running'}], 4, 'running')
+    ],
+)
+def test_check_job_status_with_connection_errors(mocker, responses, expected_times_called, expected_status):
+    """
+    given:  connection error exceptions and eventually real status.
+    when:   trying to retrieve gitlab job status
+    then:   make sure retry mechanism will be triggered on ConnectionErrors
+    """
+    side_effect_responses = []
+    for response in responses:
+        if not isinstance(response, dict):
+            side_effect_responses.append(response)
+        else:
+            r = Response()
+            r._content = json.dumps(response).encode()
+            side_effect_responses.append(r)
+
+    requests_mocker = mocker.patch.object(requests, 'get', side_effect=side_effect_responses)
```

Comment:
Why not use `requests_mock` like we do in the integrations?

Reply:
i couldn't find a way to use `side_effect` with `requests_mock`, i do not think it supports it.

if you know anything about it, dm me :)